pkg/gce-cloud-provider/compute/gce-compute.go

```
@@ -76,12 +76,12 @@ var AttachDiskBackoff = wait.Backoff{
 	Cap:      0}
 
 // WaitForOpBackoff is backoff used to wait for Global, Regional or Zonal operation to complete.
-// Default values are similar to Poll every 3 seconds with 5 minute timeout.
+// Default values are similar to Poll every 2 minutes with 6 minute timeout.
 var WaitForOpBackoff = wait.Backoff{
-	Duration: 3 * time.Second,
```

Isn't this shared with some other api calls as well? 3 seconds to 2 mins is a big change.

---

Actually I saw you change it to all the places already.

---

There's similarly named waitForOpBackoff* variables in [main.go](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/cmd/gce-pd-csi-driver/main.go#L60), I will update there as well. Otherwise we're updating all the instances where this var is used in gce-compute.go.

---

Actually, one risk I see is that "In uncommon cases, when the server is overloaded, the request might return before the default deadline is reached, or might return after zero seconds." Is there anyway to change this backoff to become constant wait until it's been 5min?

---

Good catch, do we know how this server overload error exactly looks like? If the exponential backoff conditional function can match on and swallow this error then we will get constant 2 min wait like we want. Some illustrative code: https://go.dev/play/p/iNPzsuYTMgD